src/test/java/cloud/fogbow/ras/core/processors/UnableToCheckStatusProcessorTest.java

```diff
@@ -1,313 +1,308 @@
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        order.setOrderStateInTestMode(OrderState.FAILED_AFTER_SUCCESSFUL_REQUEST);
+        this.unableToCheckStatus.addItem(order);
+
+        Mockito.doThrow(new RuntimeException()).when(this.processor).processUnableToCheckStatusOrder(order);
+
+        // exercise
+        this.thread = new Thread(this.processor);
+        this.thread.start();
+        Thread.sleep(BaseUnitTests.DEFAULT_SLEEP_TIME);
+
+        // verify
+        Mockito.verify(this.processor, Mockito.times(1)).processUnableToCheckStatusOrder(order);
+    }
+
+    // test case: When invoking the processUnableToCheckStatusOrder method with an
+    // instance that failed after a successful request, it must change the order
+    // state to FAILED_AFTER_SUCCESSFUL_REQUEST.
+    @Test
+    public void testProcessUnableToCheckStatusOrderChangeStateToFailed() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        order.setOrderStateInTestMode(OrderState.UNABLE_TO_CHECK_STATUS);
+        this.unableToCheckStatus.addItem(order);
+        
+        OrderInstance orderInstance = new ComputeInstance(BaseUnitTests.FAKE_INSTANCE_ID);
+        orderInstance.setHasFailed();
+
+        Mockito.doReturn(orderInstance).when(this.cloudConnector).getInstance(Mockito.any(Order.class));
+
+        // exercise
+        this.processor.processUnableToCheckStatusOrder(order);
+
+        // verify
+        Assert.assertEquals(OrderState.FAILED_AFTER_SUCCESSFUL_REQUEST, order.getOrderState());
+    }
+
+    // test case: When invoking the processUnableToCheckStatusOrder method and an
+    // error occurs while trying to get an instance from the cloud provider, an
+    // UnavailableProviderException will be throw.
+    @Test(expected = UnavailableProviderException.class) // Verify
+    public void testProcessUnableToCheckStatusOrderThrowsUnavailableProviderException() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        order.setOrderStateInTestMode(OrderState.UNABLE_TO_CHECK_STATUS);
+        this.unableToCheckStatus.addItem(order);
+
+        Mockito.doThrow(new UnavailableProviderException()).when(this.cloudConnector)
+                .getInstance(Mockito.any(Order.class));
+
+        // exercise
+        this.processor.processUnableToCheckStatusOrder(order);
+    }
+
+    // test case: When calling the processUnableToCheckStatusOrder method and the
+    // order instance is not found, it must change the order state to
+    // FAILED_AFTER_SUCCESSFUL_REQUEST.
+    @Test
+    public void testProcessUnableToCheckStatusOrderWithInstanceNotFound() throws FogbowException {
```

Comment:
Maybe you should improve the test case description because the situation described above allows the order to change to fulfilled state too. For last, if there is no test that do the same thing looking at fulfilled state, create it please.

Reply:
This coverage test checks if during the execution of this processor an InstanceNotFoundException has been thrown, and if this occurs the order state should be changed to FAILED_AFTER_SUCCESSFUL_REQUEST. I'll fix the test case.
And there is already a test for FULFILLED state.